### PR TITLE
#164270478 Complete user registration and email verification

### PR DESCRIPTION
**Pull request by codinger41**
#### What does this PR do?
It sets up routes and controllers for users to be able to sign up, receive emails and verify their accounts.

#### Description of Task to be completed?
Users should be able to create accounts on Authors Haven and receive an email notification containing a verification link.
#### How should this be manually tested?
- Check .env.sample for the updated .env variables.
- Run `npm run db-migrate` to run migrations.
- Run `npm run start` to start the application.
- Make a POST request to `/api/v1/user` with a JSON object containing the following fields: `email`, `username`, `name` and `password`. They're all required. If one is not sent, there'll be an error.
- If this request is successful, you should receive an email notification with a verification link to verify your account. 
- Click the link sent in the email, it's going to verify your account and return a success message.
- If the account has already been verified or not found, you should see an error.

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
#164270478
#### Questions:
N/A

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,2 +1,7 @@
 language: node_js
 script: npm run test
+node_js:
```
Comment (9jaswag) [inline .travis.yml:3]: why do we have this here?

**Comment by 9jaswag** [comment]
there are merge conflicts in this PR

File: .travis.yml (hunk 1 above)
Comment (codinger41) [reply .travis.yml:3]: @9jaswag By default, Travis was installing a very old version of node, so I had to specify it to use version 10.

File: controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        verificationId,
+        verified: false,
+      },
+    })
+      .then((updatedUser) => {
+        if (updatedUser[0] === 0) {
+          return res.status(422)
+            .json({
+              success: false,
+              message: 'Error: This user is either verified or not found.',
```
Comment (9jaswag) [inline controllers/user.js]: @leksyib Don't you think the user's verification ID should be deleted or set to NULL after they're verified? This way, the error will always be that the user doesn't exist

Comment (9jaswag) [reply controllers/user.js]: Perhaps another db hook to achieve that?

Comment (codinger41) [reply controllers/user.js]: Alright, I'll implement that.

**Comment by 9jaswag** [comment]
Looks good!

**Pull request merged**